osu.Framework/Graphics/Cursor/PopoverContainer.cs

```diff
@@ -44,20 +44,28 @@ public PopoverContainer()
         /// <returns><see langword="true"/> if a new popover was shown, <see langword="false"/> otherwise.</returns>
         internal bool SetTarget(IHasPopover? newTarget)
         {
-            target = newTarget;
-
             currentPopover?.Hide();
             currentPopover?.Expire();
 
+            if (target is Drawable oldDrawableTarget)
+                oldDrawableTarget.OnDispose -= onTargetDisposed;
+
+            target = newTarget;
+
             var newPopover = target?.GetPopover();
             if (newPopover == null)
                 return false;
 
+            if (target is Drawable newDrawableTarget)
+                newDrawableTarget.OnDispose += onTargetDisposed;
+
             AddInternal(currentPopover = newPopover);
             currentPopover.Show();
```

Comment:
very nit, but I feel it would read better to unsubscribe and subscribe to the dispose event before expire and after creation:
```suggestion
            if (target is Drawable oldDrawableTarget)
                oldDrawableTarget.OnDispose -= onTargetDisposed;

            currentPopover?.Hide();
            currentPopover?.Expire();

            target = newTarget;

            var newPopover = target?.GetPopover();
            if (newPopover == null)
                return false;

            AddInternal(currentPopover = newPopover);
            currentPopover.Show();

            if (target is Drawable newDrawableTarget)
                newDrawableTarget.OnDispose += onTargetDisposed;
```

Reply:
Why would that order matter?

Reply:
It does not matter in terms of execution or anything else obviously, just felt it would read better that way, nothing else.

Reply:
I don't follow why one would read worse than the other.

Reply:
More along the lines of `onTargetDisposed` not being necessary if the current popover isn't created yet, so it would make sense to put that after the creation.

But now that I read it again, it doesn't really matter which way to go as the subscribe logic doesn't relate itself with popovers by the naming, so you can disregard this.